test/e2e/metering_manual_install_tests.go

```diff
@@ -107,6 +108,149 @@ func testManualMeteringInstall(
+			actions:      []string{"CREATE"},
+		},
+	}
+
+	for _, manifest := range nfsManifestItems {
+
+		err := deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsServer"].resourceFile, nfsManifestItems["nfsServer"].resourceType)
+		err = deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsService"].resourceFile, nfsManifestItems["nfsService"].resourceType)
+		err = deploy.DecodeYAMLManifestToObject(nfsManifestItems["nfsStorageclass"].resourceFile, nfsManifestItems["nfsStorageclass"].resourceType)
+		if err != nil {
+			return fmt.Errorf("failed to decode the %s YAML service manifest: %v", err, manifest.resourceFile)
+		}
+
+		for _, action := range manifest.actions {
+			if action == "CREATE" {
+				ctx.Client.CoreV1().Namespaces().Create(context.Background(), namespace, metav1.CreateOptions{})
+			}
+			ctx.Logger.Debugf("Created the %s nfs_server %s nfs_service %s nfs_storageclass", nfsManifestItems["nfsServer"], nfsManifestItems["nfsService"].resourceFile, nfsManifestItems["nfsStorageclass"].resourceFile)
+			if action == "GET" {
+				_, err := ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+				if err != nil && !apierrors.IsNotFound(err) {
+					return nil
+				}
+			}
+		}
+	}
+
+	//  - Wait for service to populate spec.ClusterIP (poll.Wait functionality)
+	// poll has a closure that polls for the server IP to get populated
+	var service, err = ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+
+	// if theres an error don't poll
+	if err != nil {
+		err := wait.Poll(3*time.Second, 5*time.Minute, func() (done bool, err error) {
+			service, err = ctx.Client.CoreV1().Services(ctx.Namespace).Get(context.TODO(), "nfs-service", metav1.GetOptions{})
+			return false, err
+		}
+		// clusterIP is populated, stop polling
+		return service.Spec.ClusterIP != "", nil
+	})
+}
+
+
+
+
+
+	// 	if filepathNFSPersistentvolume.spec.nfs.server != ""
+	// 	{
+	// 	clusterIP := filepathNFSPersistentvolume.spec.nfs.server
+	// 	return clusterIP
+	// }
+	// 	return pollInternal(poller(interval, timeout), condition)
+	// }
+
+	//  - Decode the NFS PV YAML manifest
+	//get a go object
+	// after we get clusterIP, save it in a var, pvc.spec.nfs.server = val
+	filepathNFSPersistentvolume := "test/testhelpers/nfs_persistentvolume.yaml"
+	err := deploy.DecodeYAMLManifestToObject(filepathNFSPersistentvolume, &nfs_persistentvolume)
+	if err != nil {
+		return "", fmt.Errorf("failed to decode the %s YAML service manifest: %v", filepathNFSPersistentvolume, err)
+
+		// Update the NFS PV configuration w/ the populated spec.ClusterIP from the NFS Service
+		clusterIP := service.spec.nfs.server
+	}
+
+	_, err = ctx.Client.CoreV1().Namespaces().Create(context.Background(), &nfs_persistentvolume, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+	ctx.Logger.Debugf("Created the %s nfs_clusterIP", clusterIP)
```

Comment:
This can be cleaned up a bit as there's no need to decode the YAML manifest if we first get the NFS persistentvolume resource first, check if it exists, and if not, decode it, update the pvc.nfs.server to point to the svc.Spec.ClusterIP and create it. If it does exist, we don't need to do anything, at least in this current implementation.

Reply:
At this point it hasn't been decoded yet, but the 3 other manifests (server, service, storageclass) have - so I believe I do need to decode it. I think you are saying to still decode it, but within a conditional, which would make sense. Is that the thought here?

Reply:
Actually, there is not a way this could be created before because it would not have the clusterIP in it. What is a case where it would already be created?